Dynatrace/dynatrace-operator
# Re-order provisioner structure in order to fix host-monitoring

## Description

Currently the rollout of hostMonitoring fails with the error message:

MountVolume.SetUp failed for volume "osagent-storage" : rpc error: code = Unavailable desc = dynakube (....) is missing from metadata database.

This happens because if the dynakube is not needing app injection we skip adding it to the db, which we should not do, as we still have host-monitoring that relies on the CSI Driver

## How can this be tested?

Deploy operator -> Deploy Dynakube with hostMonitoring -> check if agents mount the csi volume properly and startup accordingly

## Checklist
